pulp_ansible/tests/functional/api/collection/test_sync.py

```diff
@@ -136,3 +150,42 @@ def test_sync_collection_with_stranger_version_numbers_to_check_comparisons(self
 
         content = self.cv_api.list(repository_version=f"{repo.pulp_href}versions/1/")
         self.assertGreaterEqual(len(content.results), 5)
+
+
+class FullDependenciesSync(TestCaseUsingBindings, SyncHelpersMixin):
+    """Collection sync tests for syncing collections and their dependencies."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Set up class variables."""
+        super().setUpClass()
+        cls.col_api = AnsibleCollectionsApi(cls.client)
+
+    def _clean_database(self):
```

Comment:
At some point in the future we're going to try to have tests not require the db to be in a specific state. When that happens this type of reset will be problematic. I suspect we could remove this and have the sync into the new repo demonstrate the sync is pulling in dependencies when syncing from Pulp itself. @gerrod3 wdyt? Does this idea make sense?

Reply:
Yeah, I'm in the middle of rewriting the test to use custom uploaded collections and I have removed the database cleanup code.